### PR TITLE
PARQUET-1569: [C++] Consolidate shared unit testing header files

### PR DESCRIPTION
**Pull request by wesm**
This makes the codebase slightly structurally simpler. Note there is still some code duplication around random number generation that would be good to simplify in the future

### CODE REVIEW

**Review by majetideepak — APPROVED** [review]
+1 LGTM

File: cpp/src/arrow/testing/util.h [hunk 1]
```diff
@@ -89,8 +90,8 @@ inline Status CopyBufferFromVector(const std::vector<T>& values, MemoryPool* poo
 // Sets approximately pct_null of the first n bytes in null_bytes to zero
 // and the rest to non-zero (true) values.
 ARROW_EXPORT void random_null_bytes(int64_t n, double pct_null, uint8_t* null_bytes);
-ARROW_EXPORT void random_is_valid(int64_t n, double pct_null,
-                                  std::vector<bool>* is_valid);
+ARROW_EXPORT void random_is_valid(int64_t n, double pct_null, std::vector<bool>* is_valid,
+                                  int random_seed = 0);
```
Comment (bkietz) [inline cpp/src/arrow/testing/util.h]: Probably fine: elsewhere `uint32_t` is used for seeds, while `random_seed()` returns `uint64_t`.

**Review by bkietz — APPROVED** [review]

**Comment by wesm** [comment]
Merging. The Appveyor CI failure here is spurious, seems there is some crankiness there today -- the build on my fork got to MinGW but then failed due to a connection failure to GitHub https://ci.appveyor.com/project/wesm/arrow/builds/24337857

File: cpp/src/arrow/testing/util.h (hunk 1 above)
Comment (wesm) [reply cpp/src/arrow/testing/util.h]: This was a quick fix for compatibility for one of the Parquet test generation functions, we can always change to `uint32_t` everywhere

**Pull request closed without merging**